src/main/java/org/cryptomator/integrations/autoupdate/AutoUpdateProvider.java

```diff
@@ -0,0 +1,46 @@
+package org.cryptomator.integrations.autoupdate;
+
+import org.cryptomator.integrations.common.IntegrationsLoader;
+
+import java.util.Optional;
+
+/**
+ * This is the interface used by Cryptomator to add autoupdate capabilities to the application.
+ */
+public interface AutoUpdateProvider {
+
+	/**
+	 * Loads the best-suited AutoUpdateProvider.
+	 *
+	 * @return preferred AutoUpdateProvider (if any)
+	 * @since 1.2.0
+	 */
+	static Optional<AutoUpdateProvider> get() {
+		return IntegrationsLoader.load(AutoUpdateProvider.class);
+	}
+
+	/**
+	 * Initializes autoupdate features for the application. This sets configuration needed for autoupdate to work.
+	 * The best place to do it is as soon after startup as possible, but no sooner than the app's main window is shown.
+	 */
+	void initAutoUpdate();
+
+	/**
+	 * Perform cleanups on the autoupdate features when the application exits.
+	 */
+	void cleanUpAutoUpdate();
+
+	/**
+	 * Enable autoupdate features for the applcation.
+	 *
+	 * @return True, in case enabling the features was successful, false otherwise.
+	 */
+	boolean enableAutoUpdate();
```

Comment:
Only returning a boolean does not give a deeper insight if enabling it it fails. I suggest to refactor the method:
```java
void enableAutoUpdate() throws AutoUpdateException;
```
Calling it multiple times when already enabled should not throw an exception (aka making the function idempotent).

The documentation should be updated also then.

Reply:
You are right, I'll change that.